src/ring.rs

```diff
@@ -140,6 +140,17 @@ impl<'a> LogEntryRingBuffer<'a> {
         }
     }
 
+    /// Read a single entry from the RingBuffer tail (oldest) without advancing
+    /// the read cursor
+    #[inline]
+    pub fn peek_entry(&mut self) -> Option<LogEntry> {
```

Comment:
Does this need to be `&mut` access to self?

Reply:
Nope, b845234b8a7afbea06d4f377e2d08cc44db39618